### ISSUE TITLE
Categorical should normalize ps or warn if not normalized

### ISSUE THREAD

**Issue by longouyang** [probmods/webppl#686]
Webchurch normalized the probability arguments to its categorical distribution; people used to that will be surprised by things like:

``` js
var d = Categorical({ps: [1, 1], vs: ['a','b']});
d.score('b'); // returns 0, not -0.69
```

I think we should either normalize for the user or warn if the arguments don't (approximately) sum to 1.

More generally, we should check input arguments to base distributions.

**Comment by hawkrobe** [comment]
I think this is a duplicate of #397 and #582

**Comment by null-a** [comment]
> I think this is a duplicate of #582

I've now closed #582.

> More generally, we should check input arguments to base distributions.

Indeed. #332.
